Automattic/wp-calypso
# Add ProductCard promo nudge sub-component

As part of the work related to p1HpG7-87V-p2, this PR adds a promo nudge sub-component to the `ProductCard` and makes use of it inside the `ProductsSelector`.

This is the 4th and final PR that implements the promo nudge in Calypso. The previous PRs are:
- #38311 
- #38337 
- #38344 

Here are the **after** states for Calypso and Jetpack Connect flow:

<img width="576" alt="Screenshot 2019-12-12 at 14 03 39" src="https://user-images.githubusercontent.com/478735/70714436-72121700-1ce8-11ea-9d9f-2219a576de51.png">
<img width="556" alt="Screenshot 2019-12-12 at 14 04 16" src="https://user-images.githubusercontent.com/478735/70714438-72aaad80-1ce8-11ea-897d-aff3b3941a00.png">

#### Changes proposed in this Pull Request

* Add `ProductCardPromoNudge` sub-component
* Update devdocs example file
* Update `ProductCard` readme file

#### Testing instructions



* Go to http://calypso.localhost:3000/plans/ and select a Jetpack site having no Jetpack Backup product
* Confirm that you can see the promo nudge inside the card, as per the designs
* Purchase one of the Jetpack Backup products and confirm the nudge is no longer there
* Go to Jetpack Connect flow: http://calypso.localhost:3000/jetpack/connect/store
* Confirm that the promo nudge is inside the card, as per the designs
* Go to devdocs: http://calypso.localhost:3000/devdocs/design/product-card and confirm that one of the examples presents the promo nudge and that the readme file is updated

Fixes n/a
